### PR TITLE
Remove SqlClient dependency from SqlClientCompatibility packages

### PR DESCRIPTION
**Pull request by nyctef**
context: https://github.com/red-gate/SQLCompareEngine/issues/5256

tl;dr we're trying to remove the need for separate `.SSMS` packages across as much of the stack as possible, particularly now that we think we would need additional `.SSMS20` packages which reference a specific v5 version of SqlClient.

From what I can tell, these packages were forcing consumers to reference a specific version of SqlClient, but bringing in source code that could be built against any version of SqlClient.

This will need more testing in at least SqlCompareEngine before it can be merged, though.

Outstanding questions:
- [x] is this code actually relevant for SqlCompareEngine, or is it actually just needed at whichever point is going to inject a connection into SqlCompareEngine? There might be some required refactoring that makes this change less necessary.
   - Answer: it looks like the relevant usage is in a `ConnectionProperties` class which is only really used by consumers rather than the engine itself - however moving it may well be tricky

### CODE REVIEW

**Comment by samblackburn** [comment]
What is the main responsibility of the `SqlClientCompatibility` packages, now that they don't rely on `SqlClient`?

**Comment by samblackburn** [comment]
OK think I've got it. So the nuspec no longer has a MDS reference (because that would constrain the consumer to a specific version).
The csproj still has a reference (so that you can build the MicroLibraries solution locally) but it's a source package, so the compilation output doesn't affect the nuget package.
